File: swift/Tests/2025Tests/2025.10.test.swift
import Testing
@testable import AdventOfCode

struct Year2025Day10Tests {
  let day = Year2025.Day10(challengeYear: 2025)

  @Test("Day 10, Part 1") func part1() {
    #expect(day.part1() as? Int == 491)
  }

  @Test("Day 10, Part 2") func part2() {
    #expect(day.part2() as? Int == 20617)
  }
}

File: js/2025/tests/2025-10.test.js
import { loadInput } from '../../utilities/io.js';
import { part1 } from '../10/part-1.js';
import { part2 } from '../10/part-2.js';

const { lines, data } = loadInput(2025, 10);
let grid;

describe('2025 Day 10', () => {
  beforeEach(() => (grid = lines.map((line) => line.split(''))));

  test('Part 1', () => {
    expect(part1({ lines, grid, data })).toBe(491);
  });

  test('Part 2', () => {
    expect(part2({ lines, grid, data })).toBe(20617);
  });
});
